### DISCUSSION TITLE
User Presence

### DISCUSSION THREAD

**Discussion by pragmatech** [pocketbase/pocketbase#845] [Q&A] [answered]
Issues central to user presence have been discussed elsewhere https://github.com/pocketbase/pocketbase/discussions/467. However, I thought a generalized discussion wouldn't hurt. 

In my socket.io efforts, listing the room's participants has worked, albeit with something not realtime regarding websockets'heartbeats, docker, perhaps vdom, and who knows what else. A couple years ago, Supabase's CEO kindly replied: "Presence is surprisingly difficult. It is something that we can tackle with our Elixir server". 
...and indeed Elixir seems a sort of magical beast regarding realtime.

Is pursuit of a rock-solid user presence indication, within the refreshingly minimal and sovereign pocketbase, barking up the wrong tree? Any scoops or pointers are most welcome. thanks.

**Comment by ganigeorgiev** [comment, marked as answer]
User presence in PocketBase is hard not because it is hard to implement, but because it is not clear who will have access to the presence information and whether there will be some "grouping" (as in your case with the room participants).

With https://github.com/pocketbase/pocketbase/issues/468, we will add additional realtime event hooks, but I don't plan adding a dedicated API and it will be up to the developers to decide how to use the hooks and who will have access to the presence information - only admins, only subscribers to a specific topic or some other criteria.
